File: lista java/at6.java

class Motor {
    private String tipo;

    
    public Motor(String tipo) {
        this.tipo = tipo;
    }

    
    public String getTipo() {
        return tipo;
    }
}


class Carro {
    private String marca;
    private String modelo;
    private Motor motor;  

    
    public Carro(String marca, String modelo, Motor motor) {
        this.marca = marca;
        this.modelo = modelo;
        this.motor = motor;
    }

    
    public void exibirDetalhes() {
        System.out.println("Carro: " + marca + " " + modelo);
        System.out.println("Motor: " + motor.getTipo());
    }

    public static void main(String[] args) {
        
        Motor motorV8 = new Motor("V8");

        
        Carro carro = new Carro("Ford", "Mustang", motorV8);

        
        carro.exibirDetalhes();
    }
}

File: lista Go/at6.go
package main

import "fmt"


type Motor struct {
    Tipo string
}


type Carro struct {
    Marca  string
    Modelo string
    Motor  Motor  
}


func (c Carro) ExibirDetalhes() {
    fmt.Printf("Carro: %s %s\n", c.Marca, c.Modelo)
    fmt.Printf("Motor: %s\n", c.Motor.Tipo)
}

func main() {
    
    motorV8 := Motor{Tipo: "V8"}

    
    carro := Carro{Marca: "Ford", Modelo: "Mustang", Motor: motorV8}

    
    carro.ExibirDetalhes()
}
